theano/gof/graph.py

```diff
@@ -524,6 +526,14 @@ def eval(self, inputs_to_values=None):
     def __getstate__(self):
         d = self.__dict__.copy()
         d.pop("_fn_cache", None)
+        if (not config.pickle_test_value) \
+           and (hasattr(self.tag, 'test_value')) \
+           and (not type(config).pickle_test_value.is_default):
```

Comment:
This is not good. There should be 2 if. If the flag is True, but not the default, we should not print the warning. So if should be something like:
```
if (not config.pickle_test_value and
          hasattr(self.tag, 'test_value')):
    if type(config).pickle_test_value.is_default:
        warnings...
    t = ...
```

Reply:
Do I need to add another pull request to fix this?

Reply:
This might be a good thing, yes

Reply:
It's fixed in #6366